src/tests/ftest/util/systemctl_utils.py

```diff
@@ -0,0 +1,249 @@
+    status_keys = ["reset-failed", "stop", "disable"]
+    mapping = {"stop": "active", "disable": "enabled", "reset-failed": "failed"}
+    check_hosts = NodeSet(hosts)
+    loop = 1
+    # Reduce 'max_loops' to 2 once https://jira.hpdd.intel.com/browse/DAOS-7809 has been resolved
+    max_loops = 3
+    while check_hosts:
+        # Check the status of the service on each host
+        result = get_service_status(logger, check_hosts, service)
+        check_hosts = NodeSet()
+        for key in status_keys:
+            if result[key]:
+                if loop == max_loops:
+                    # Exit the while loop if the service is still running
+                    logger.error(
+                        " - Error %s still %s on %s", service, mapping[key], result[key])
+                    result["status"] = False
+                else:
+                    # Issue the appropriate systemctl command to remedy the
+                    # detected state, e.g. 'stop' for 'active'.
+                    command = command_as_user(get_systemctl_command(key, service, user), user)
+                    run_remote(logger, result[key], command)
+
+                    # Run the status check again on this group of hosts
+                    check_hosts.add(result[key])
+        loop += 1
+
+    return result["status"]
+
+
+def get_systemctl_command(unit_command, service, user="root"):
+    """Get the systemctl command for the specified inputs.
+
+    Args:
+        unit_command (str): command to issue for the service
+        service (str): service for which to issue the command
+        user (str, optional): user to use to issue the command. Defaults to "root".
+
+    Returns:
+        str: the systemctl command for the specified service and user
+    """
+    command = ["systemctl"]
+    if user != "root":
+        command.append(f"--user {user}")
+    if unit_command:
+        command.append(unit_command)
+    if service:
+        command.append(service)
+    return " ".join(command)
+
+
+def get_service_file(logger, hosts, service, user, verbose=True, timeout=120):
+    """Get the service file.
+
+    Args:
+        logger (Logger): logger for the messages produced by this method
+        hosts (NodeSet): hosts on which to run the command
+        service (str): service for which to issue the command
+        user (str, optional): user to use to issue the command. Defaults to "root".
+        verbose (bool, optional): log the command output. Defaults to True.
+        timeout (int, optional): number of seconds to wait for the command to complete.
+            Defaults to 120 seconds.
+
+    Raises:
+        SystemctlFailure: if there is a problem obtaining the service file
+
+    Returns:
+        str: the service file
+    """
+    command = ' | '.join([
+        get_systemctl_command("status", service, user),
+        "grep 'Loaded:'",
+        "grep -oE '/.*service'",
+        "xargs -r sh -c '[ -e \"$0\" ] && echo \"$0\"'"
+    ])
```

Comment:
There are a couple bugs here from my code. We need to redirect stderr to stdout in case the file doesn't exist. And we shouldn't use `xargs -r` since if `grep` doesn't find anything we want to fail.
```suggestion
    command = ' | '.join([
        get_systemctl_command("status", service, user) + " 2>&1",
        "grep 'Loaded:'",
        "grep -oE '/.*service'",
        "xargs sh -c '[ -e \"$0\" ] && echo \"$0\"'"
    ])
```

E.g. this is for the case of
```
$ systemctl status daos_agent.service
Unit daos_agent.service could not be found.
```

Reply:
Implemented